src/python/pants/option/subsystem.py

```diff
@@ -100,6 +100,10 @@ def register_options(cls, register):
 
         Subclasses may override and call register(*args, **kwargs).
         """
+        for attrname in dir(cls):
```

Comment:
I don't remember if this will collect all the attributes in the class hierarchy, gotta check.

Reply:
It will.